canvas_modules/common-canvas/src/common-properties/controls/expression/expression-builder/expression-builder.jsx

```diff
@@ -31,85 +31,86 @@ export default class ExpressionBuilder extends React.Component {
 		this.onChange = this.onChange.bind(this);
 		this.onBlur = this.onBlur.bind(this);
 		this.onSelectionChange = this.onSelectionChange.bind(this);
-		this.lastCursorPos = null;
-
 	}
 
 	onChange(newValue) {
 		const value = (typeof newValue === "string") ? newValue : newValue.toString();
-		let cursor = this.editor.getCursor();
+		const somethingSelected = this.editor.viewState.state.selection.ranges.some((r) => !r.empty);
+		let cursor = this.editor.viewState.state.selection.main.head;
```

Comment:
Are both of these internal to CodeMirror?  We probably should add optional chaining in case CodeMirror changes the code and these objects don't exist.

Reply:
I will add a new method which will return `this.editor?.viewState?.state`

Getting the `state.doc`, `state.selection`, `state.replaceSelection` is documented in this migration guide - https://codemirror.net/docs/migration/